chain/x/oracle/keeper/infractions.go

```diff
@@ -0,0 +1,66 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/slashing"
+
+	"github.com/bandprotocol/bandchain/chain/x/oracle/types"
+)
+
+// HandleValidatorReport handles a validator report, must be called once per validator per request.
+func (k Keeper) HandleValidatorReport(ctx sdk.Context, requestID types.RequestID, address sdk.ValAddress, reported bool) {
+	logger := k.Logger(ctx)
+
+	// If validator not found or has been jailed, we will reset consecutive miss count if report info exists.
+	validator := k.StakingKeeper.Validator(ctx, address)
+	if validator == nil || validator.IsJailed() {
+		// Validator was (a) not found or (b) already jailed
+		reportInfo, err := k.GetValidatorReportInfo(ctx, address)
+		if err == nil {
+			reportInfo.ConsecutiveMissed = 0
+			k.SetValidatorReportInfo(ctx, address, reportInfo)
+		}
+		logger.Info(
+			fmt.Sprintf("Validator %s missed report, but was either not found in store or already jailed", address),
+		)
+		return
+	}
+	// fetch report info
+	reportInfo, err := k.GetValidatorReportInfo(ctx, address)
+	if err != nil {
```

Comment:
While it's true that here we know that the only possible error is not found, assuming `err != nil` will always be that way is antipattern. We should use `k.HasValidatorReportInfo` in my opinion. Or make `GetValidatorReportInfoWithDefault` function that returns empty report info if not exist